tests/test_nblist.py

```diff
@@ -128,9 +131,70 @@ def test_neighborlist():
                         print("TESTING bidx", bidx)
                         print(sorted(a))
                         print(sorted(b))
-                    # print(a, b)
                     np.testing.assert_equal(sorted(a), sorted(b))
 
-                # np.testing.assert_equal(ref_ixn_list, test_ixn_list)
 
-                # output is a group of 32
+def test_neighborlist_ligand_host():
+    ff = hif2a_ligand_pair.ff
+    ligand = hif2a_ligand_pair.mol_a
+    ligand_coords = get_romol_conf(ligand)
+
+    system, host_coords, box, top = build_water_system(4.0)
+    num_host_atoms = host_coords.shape[0]
+    host_coords = np.array(host_coords)
+
+    minimized_coords = minimize_host_4d([ligand], system, host_coords, ff, box)
+    coords = np.concatenate([minimized_coords, ligand_coords])
+
+    N = coords.shape[0]
+    D = 3
+    cutoff = 1.0
+    block_size = 32
+    padding = 0.1
+
+    np.random.seed(1234)
+    diag = np.amax(coords, axis=0) - np.amin(coords, axis=0) + padding
+    box = np.eye(3)
+    np.fill_diagonal(box, diag)
+
+    # Can only sort the host coords, but not the row/ligand
+    sort = True
+    if sort:
+        perm = hilbert_sort(coords[:num_host_atoms] + np.argmin(coords[:num_host_atoms]), D)
+        coords[:num_host_atoms] = coords[:num_host_atoms][perm]
+
+    col_coords = np.expand_dims(coords[:num_host_atoms], axis=0)
+    # Compute the reference interactions of the ligand
+    ref_ixn_list = []
+    num_ligand_atoms = coords[num_host_atoms:].shape[0]
+    num_blocks_of_32 = (num_ligand_atoms + block_size - 1) // block_size
+    box_diag = np.diag(box)
+    for rbidx in range(num_blocks_of_32):
+        row_start = num_host_atoms + (rbidx * block_size)
+        row_end = min(num_host_atoms + ((rbidx + 1) * block_size), N)
+        row_coords = coords[row_start:row_end]
+        row_coords = np.expand_dims(row_coords, axis=1)
+        deltas = row_coords - col_coords
+        deltas -= box_diag * np.floor(deltas / box_diag + 0.5)
```

Comment:
Nice -- could also export this idiom to `delta_r` in `jax_utils` https://github.com/proteneer/timemachine/blob/1fb7f3804d94d245c3b4c284d5860e60b7ff851e/timemachine/potentials/jax_utils.py#L94-L95 , avoiding a Python for-loop